Add shim for getCurrentBlasHandle

janeyx99 opened the pull request:
1. Start a cuda shim_common.cpp
2. Add a getCurrentBlasHandle shim, which is needed to support vLLM cuda kernels.

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #168276

janeyx99 committed:
Add shim for getCurrentBlasHandle

[ghstack-poisoned]

eqy reviewed (commented):
dumb question: what's the intended use-case for this? So extensions can reuse PyTorch's cuBLAS handle (and workspace?)?

janeyx99 commented:
@eqy not a dumb q at all! It's used in some cuda kernels in vllm: https://github.com/search?q=repo%3Avllm-project%2Fvllm%20getCurrentCUDABlasHandle&type=code 

if this isn't the right way their code should use the API, please holler!

eqy reviewed (approved):
If they're already functionally using something like this and it ostensibly works then that makes me less nervous.
Since the get handle function also implicitly sets the workspace that seems fine as well...

albanD reviewed (approved):
Ok!

janeyx99 committed:
Update on "Add shim for getCurrentBlasHandle"

1. Start a cuda shim_common.cpp
2. Add a getCurrentBlasHandle shim, which is needed to support vLLM cuda kernels.




[ghstack-poisoned]

janeyx99 committed:
Update on "Add shim for getCurrentBlasHandle"

1. Start a cuda shim_common.cpp
2. Add a getCurrentBlasHandle shim, which is needed to support vLLM cuda kernels.




[ghstack-poisoned]

janeyx99 commented:
@pytorchbot merge
